### PR TITLE
Data table pagination ajax

### PR DESCRIPTION
**Pull request by itkg-alavieille**
in relation with : 
https://github.com/open-orchestra/open-orchestra-base-api-bundle/pull/17
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/315
https://github.com/open-orchestra/open-orchestra-model-interface/pull/71
https://github.com/open-orchestra/open-orchestra-user-bundle/pull/56
https://github.com/open-orchestra/open-orchestra-workflow-function-bundle/pull/6

### CODE REVIEW

File: ApiBundle/Controller/ContentController.php [hunk 1]
```diff
@@ -101,10 +101,32 @@ public function showOrCreateAction(Request $request, $contentId)
     public function listAction(Request $request)
     {
         $contentType = $request->get('content_type');
+        $columns = $request->get('columns');
```
Comment (nicolasThal) [inline ApiBundle/Controller/ContentController.php]: isn't there a way to factorize the creation of all those parameters?

File: ApiBundle/Facade/PaginateCollectionFacade.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * @Serializer\Type("integer")
+     * @Serializer\SerializedName("recordsFiltered")
+     */
+    public $recordsFiltered;
+
+    /**
+     * @param $recordsTotal
+     */
+    public function setRecordsTotal($recordsTotal)
```
Comment (nicolasThal) [inline ApiBundle/Facade/PaginateCollectionFacade.php]: as the properties are public and they are integer, you do not need a setter to set them

File: LogBundle/Facade/LogCollectionFacade.php [hunk 3]
```diff
@@ -36,4 +48,20 @@ public function getLogs()
     {
         return $this->logs;
     }
+
+    /**
+     * @param $recordsTotal
+     */
+    public function setRecordsTotal($recordsTotal)
```
Comment (nicolasThal) [inline LogBundle/Facade/LogCollectionFacade.php]: do not put setters

**Comment by nicolasThal** [comment]
there is a lot of code duplication.

I think that there would be a way to factorize it

**Comment by itkg-alavieille** [comment]
@nicolasThal

File: ApiBundle/Controller/ContentController.php (hunk 1 above)
Comment (nicolasThal) [inline ApiBundle/Controller/ContentController.php]: did you forgot to refactor?

**Comment by nicolasThal** [comment]
you still can suppress the setter in the facade and directly give the value to the property : $facade->count = $count

File: ApiBundle/Controller/ContentController.php (hunk 1 above)
Comment (itkg-alavieille) [inline ApiBundle/Controller/ContentController.php]: The contentController, ContentController and SiteController use the different method in repository which add additional filters so i can't use the trait HandleRequestDataTable

Comment (nicolasThal) [inline ApiBundle/Controller/ContentController.php]: Are you sure?

You could give as a parameter the name of the method in the repository for example

Comment (itkg-alavieille) [inline ApiBundle/Controller/ContentController.php]: Yes but this method take orther parameter. for instance in the contentController the method for pagination take a parameter contentType.

Comment (nicolasThal) [inline ApiBundle/Controller/ContentController.php]: all right.

So in this case, I would only suggest to refactor the extraction of all the parameter of the request in an external method

**Comment by itkg-alavieille** [comment]
@nicolasThal

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**